app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala

```diff
@@ -1115,54 +1116,73 @@ class SectionRenderingService(
+            .evalAndApplyTypeInfoFirst(e)
+            .stringRepresentation
+          List(
+            SummaryListRowHelper.summaryListRow(
+              key.map(sse(_, false)).getOrElse(fcrd.label(formComponent)),
+              Html(expStr),
+              Some(""),
+              "",
+              "",
+              "",
+              List()
+            )
           )
-        )
-      case MiniSummaryList.Row(key, MiniSummaryListValue.Reference(FormCtx(formComponentId)), _) =>
-        val formModel = ei.formModelOptics.formModelVisibilityOptics.formModel
-        formModel.sectionNumberLookup
-          .get(formComponentId)
-          .map { sn =>
-            val sectionTitle4Ga = sectionTitle4GaFactory(formModel.pageModelLookup(sn), sn)
-            val fc = formModel.fcLookup.get(formComponentId).get
-            val fcUpdated = key.map(k => fc.copy(shortName = Some(k))).getOrElse(fc)
-            FormComponentSummaryRenderer
-              .summaryListRows[DataOrigin.Mongo, SummaryRender](
-                fcUpdated,
-                ei.singleton.page.id.map(_.modelPageId),
-                formTemplateId,
-                ei.formModelOptics.formModelVisibilityOptics,
-                ei.maybeAccessCode,
-                sn,
-                sectionTitle4Ga,
-                obligations,
-                validationResult,
-                ei.envelope,
-                ei.addressRecordLookup,
-                None,
-                FastForward.StopAt(ei.sectionNumber)
-              )
-          }
-          .toList
-          .flatten
-    }.flatten
-    new GovukSummaryList()(SummaryList(slRows))
+        case MiniSummaryRow.ValueRow(key, MiniSummaryListValue.Reference(FormCtx(formComponentId)), _) =>
+          val formModel = ei.formModelOptics.formModelVisibilityOptics.formModel
+          formModel.sectionNumberLookup
+            .get(formComponentId)
+            .map { sn =>
+              val sectionTitle4Ga = sectionTitle4GaFactory(formModel.pageModelLookup(sn), sn)
+              val fc = formModel.fcLookup.get(formComponentId).get
+              val fcUpdated = key.map(k => fc.copy(shortName = Some(k))).getOrElse(fc)
+              FormComponentSummaryRenderer
+                .summaryListRows[DataOrigin.Mongo, SummaryRender](
+                  fcUpdated,
+                  ei.singleton.page.id.map(_.modelPageId),
+                  formTemplateId,
+                  ei.formModelOptics.formModelVisibilityOptics,
+                  ei.maybeAccessCode,
+                  sn,
+                  sectionTitle4Ga,
+                  obligations,
+                  validationResult,
+                  ei.envelope,
+                  ei.addressRecordLookup,
+                  None,
+                  FastForward.StopAt(ei.sectionNumber)
+                )
+            }
+            .toList
+            .flatten
+        case MiniSummaryRow.HeaderRow(header) => ???
```

Comment:
Nope

Reply:
oops

Reply:
I will throw Exception here as `aux` method does't expect `HeaderRow` in the list of MiniSummaryRow

Reply:
done